json_encoder.go

```diff
@@ -104,6 +105,18 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 // grade-school notation otherwise).
 func (enc *jsonEncoder) AddFloat64(key string, val float64) {
 	enc.addKey(key)
+	if math.IsNaN(val) {
+		enc.bytes = append(enc.bytes, `"NaN"`...)
```

Comment:
I'm fine with `key-error`, but let's defer that to another change - there are a few places where we handle errors like this. Created #43.